xapian-core/cluster/cluster.cc

```diff
@@ -295,11 +296,29 @@ Centroid::recalc_magnitude()
 	magnitude += it->second * it->second;
 }
 
-Cluster::Cluster()
+void
+Cluster::operator=(const Cluster &other)
+{
+    // pointers are reference counted.
+    internal = other.internal;
+}
```

Comment:
`operator=` should really return a reference to the object - so return type `Cluster&` (in this case) and `return *this;`